tests/prov/test_prov_k8s_cortx_upgrade.py

```diff
@@ -372,3 +372,113 @@ def test_42179(self):
+            worker_thread.start()
+            pull_image_thread_list.append(worker_thread)
+        for each in pull_image_thread_list:
+            each.join()
+        que = multiprocessing.Queue()
+        start_upgrade_proc = multiprocessing.Process(target=self.rolling_upgrade, args=(
+            True, que, self.prov_conf["upgrade_start"]))
+        # start the Upgrade
+        LOGGER.info("Step 2: Start the upgrade.")
+        start_upgrade_proc.start()
+        time.sleep(self.prov_conf["sleep_time"])  # Wait to start upgrade_thread thread
+        # Verify the upgrade status.
+        LOGGER.info("Step 3: Verify the upgrade status.")
+        status_resp = self.get_status()
+        LOGGER.info("Status is %s", status_resp)
+        # suspend the upgrade.
+        LOGGER.info("Step 4: Suspend the upgrade.")
+        suspend_resp = self.upgrade_obj.upgrade_software(self.master_node_list[0],
+                                                            self.prov_conf['k8s_dir'],
+                                                            exc=False, flag=
+                                                            self.prov_conf["upgrade_suspend"])
+        assert_utils.assert_true(suspend_resp)
+        assert_utils.assert_in(cons.UPGRADE_SUSPEND_MSG, suspend_resp[1])
+        # verify the upgrade status
+        status_resp = self.get_status()
+        LOGGER.info("Status is %s", status_resp)
+        # Verify the already paused process by giving pause command
+        LOGGER.info("Step 5: verify the already suspended status.")
+        suspended_resp = self.upgrade_obj.upgrade_software(self.master_node_list[0],
+                                                            self.prov_conf['k8s_dir'],
+                                                            exc=False, flag=
+                                                            self.prov_conf["upgrade_suspend"])
+        assert_utils.assert_true(suspended_resp)
+        assert_utils.assert_in(cons.UPGRADE_SUSPENDED_MSG, suspended_resp[1])
+        time.sleep(self.prov_conf["sleep_time"])
+        start_upgrade_proc.terminate()
+        # Verify the POD and Service status
+        LOGGER.info("Step 6: Verify the PODs and Services status.")
+        pod_status = self.upgrade_obj.prov_obj.check_pods_status(self.master_node_list[0])
+        assert_utils.assert_true(pod_status)
+        resp = self.upgrade_obj.prov_obj.check_service_status(self.master_node_list[0])
+        assert_utils.assert_true(resp)
+        installed_version = self.upgrade_obj.prov_obj.get_installed_version(
+            self.master_node_list[0])
+        LOGGER.info("Upgraded to version %s", installed_version)
+        self.collect_sb = False
+        LOGGER.info("--------- Test Completed ---------")
+
+    @pytest.mark.lc
+    @pytest.mark.cortx_upgrade
+    @pytest.mark.tags("TEST-41952")
+    def test_41952(self):
+        """Verify verious stages upgrade status."""
```

Comment:
various* 
Please correct the statement to 'Verify upgrade status for various upgrade states.(start | suspend | resume)'